RocketChat/Rocket.Chat.ReactNative
# [FIX] Navigate to team from directory

## Proposed changes
After we merged this [PR](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/3823), we introduced an error that makes navigation from directory to teams impossible.

## Issue(s)	
N/A

## How to test or reproduce
On chats list, click on the global icon to navigate to directory view, then change the filter to teams and click on a team to navigate to it.

Check if the navigation to channels and users still works fine.

Regression tests (related to the original PR) 
Open a chat:

* check if the messages are loading;
* check if the messages are updated;
* check the threads;
* check the rooms info;
* Check if the list of chats is still working;



## Screenshots

| Before | After |
|-------|-------|
| <image src="https://user-images.githubusercontent.com/14894380/159788936-c8e95652-5ee8-4131-84bf-a17a3baa7310.mov" /> | <image src="https://user-images.githubusercontent.com/14894380/159789127-8c22b256-d13a-4a20-9576-8a485144a11f.mov " /> | 




## Types of changes




## Checklist



## Further comments
N/A
